core/domain/suggestion_registry.py

```diff
@@ -355,16 +418,31 @@ def __init__(
         self.target_id = target_id
         self.target_version_at_submission = target_version_at_submission
         self.author_id = author_id
-        self.change = exp_domain.ExplorationChange(change)
+        self.change: exp_domain.ExplorationChange = (
+            exp_domain.ExplorationChange(change)
+        )
         self.score_category = score_category
-        self.language_code = language_code
-        self.last_updated = last_updated
+        # In BaseSuggestion, language_code is defined with only string type
+        # but here language_code is of Optional[str] type because language_code
+        # can accept None values as well. So, due to this conflict in types
+        # MyPy throws an `Incompatible types in assignment` error. Thus to
+        # avoid the error, we used ignore here.
+        self.language_code = language_code  # type: ignore[assignment]
+        # In BaseSuggestion, last_updated is defined with only datetime type
+        # but here last_updated is of Optional[datetime] type because
+        # last_updated can accept None values as well. So, due to this conflict
+        # in types MyPy throws an `Incompatible types in assignment` error.
+        # Thus to avoid the error, we used ignore here.
+        self.last_updated = last_updated  # type: ignore[assignment]
         self.edited_by_reviewer = edited_by_reviewer
-        # Currently, we don't allow adding images in the "edit state content"
-        # suggestion, so the image_context is None.
-        self.image_context = None
-
-    def validate(self):
+        # In BaseSuggestion, image_context is defined as string type attribute
+        # but currently, we don't allow adding images in the "edit state
+        # content" suggestion, so the image_context is None here and due to
+        # None MyPy throws an `Incompatible types in assignment` error.
+        # Thus to avoid the error, we used ignore here.
+        self.image_context = None  # type: ignore[assignment]
```

Comment:
Should we change the definitions in the base type?

Reply:
I think we should not do this!!. because `self.image_context` can be of Optional[str] type in this class only and in other classes ( majority of the classes ) it has a string type. And in the base class also, we are using `self.image_context` as str type not Optional[str] so in methods where it is used as a string, we have to assert there to narrow down its type.

Reply:
Makes sense.